### PR TITLE
VSL-91: Added a pre check if signer already exists before adding

### PR DESCRIPTION
**Pull request by KsenijaZivkovic**
- This check is triggered when the action is executed
- Added check on UI against adding the same user to the treasury

### CODE REVIEW

File: packages/client/src/components/SafeSettings.js [hunk 1]
```diff
@@ -266,9 +265,14 @@ const AddSafeOwner = ({ web3, onCancel, onNext }) => {
 
   const onAddressChange = async (newAddress) => {
     setAddress(newAddress);
-    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)));
+    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)) && !isAddressExisting(safeOwners, newAddress));
   };
 
+  const isAddressExisting = (safeOwners, newAddress) => {
+    const address = newAddress.startsWith("0x") ? newAddress : `0x${newAddress}`;
```
Comment (amyliu6) [inline packages/client/src/components/SafeSettings.js:272]: we can use this `formatAddress` defined here: https://github.com/DapperCollectives/VESSEL/blob/main/packages/client/src/utils.js#L45

Comment (KsenijaZivkovic) [reply packages/client/src/components/SafeSettings.js]: Nice catch, updated!

File: packages/client/src/components/SafeSettings.js [hunk 2]
```diff
@@ -193,7 +193,6 @@ const EditSignatureThreshold = ({
 
 const RemoveSafeOwner = ({ web3, safeOwner, onCancel, onSubmit }) => {
   const { isAddressValid } = useAddressValidation(web3.injectedProvider);
-  const [name, setName] = useState(safeOwner.name);
   const [address, setAddress] = useState(safeOwner.address);
```
Comment (linxiao-li) [inline packages/client/src/components/SafeSettings.js:196]: Good job removing the unnecessary name state here!

File: packages/client/src/components/SafeSettings.js [hunk 3]
```diff
@@ -266,9 +265,14 @@ const AddSafeOwner = ({ web3, onCancel, onNext }) => {
 
   const onAddressChange = async (newAddress) => {
     setAddress(newAddress);
-    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)));
+    setAddressValid(isAddr(newAddress) && (await isAddressValid(newAddress)) && !isAddressExisting(safeOwners, newAddress));
   };
 
+  const isAddressExisting = (safeOwners, newAddress) => {
+    const address = formatAddress(newAddress);
+    return safeOwners.filter(obj => obj.address === address).length !== 0;
```
Comment (linxiao-li) [inline packages/client/src/components/SafeSettings.js:273]: We should make sure the `verified` field gets evaluated as well
```suggestion
    return safeOwners.filter(obj => obj.address === address && obj.verified).length !== 0;
```

Comment (KsenijaZivkovic) [reply packages/client/src/components/SafeSettings.js]: Added!

**Review by jacksonConrad — APPROVED** [review]

**Review by linxiao-li — APPROVED** [review]

**Pull request merged**